core/trino-main/src/main/java/io/trino/operator/output/UnnestingPositionsAppender.java

```diff
@@ -211,6 +211,14 @@ public long getSizeInBytes()
                 (rleValue != null ? rleValue.getSizeInBytes() : 0);
     }
 
+    void addSizesToAccumulator(PositionsAppenderSizeAccumulator accumulator)
```

Comment:
I think this could just return a record (and possibly `OptionalLong` for directSizeInBytes). IMO it would be cleaner and easier to understand than accumulator.

Reply:
I don’t think the extra allocation per appender per check is worthwhile for the small bit of extra clarity is worthwhile here. At that point, you can’t box the direct size into an optional because RLE’s have both a “size” and a separate “direct size” value that need to be summed independently from each other.